prov/gni/src/gnix_mr_cache.c

```diff
@@ -943,6 +1268,10 @@ static int __mr_cache_search_stale(
 	gnix_mr_cache_key_t *mr_key;
 	gnix_mr_cache_entry_t *mr_entry, *tmp;
 
+	if (cache->attr.lazy_deregistration) {
+		__clear_notifier_events(cache);
```

Comment:
I think you're right here.  I have a mental note to reevaluate how often we check for events, but maybe it would be good to check even without lazy deregistration  sometimes.  I'll create an issue so we can discuss.